src/realm/impl/simulated_failure.cpp

```diff
@@ -175,11 +175,11 @@ bool SimulatedFailure::do_check_trigger(FailureType failure_type) noexcept
     return false;
 }
 
-#if (REALM_ARCHITECTURE_X86_32 && REALM_IOS) || (REALM_ARCHITECTURE_X86_64 && REALM_WATCHOS)
+#if (REALM_ARCHITECTURE_X86_32 && REALM_IOS) || (REALM_ARCHITECTURE_X86_32 && REALM_WATCHOS) || (REALM_ARCHITECTURE_X86_64 && REALM_WATCHOS) || (REALM_ARCHITECTURE_ARM_64 && REALM_WATCHOS)
```

Comment:
This is a really confusing condition. Is it just `(REALM_WATCHOS && !REALM_ARCHITECTURE_ARM64_32)` or is there something else going on?

Reply:
We previously added a condition to support old intel simulators REALM_ARCHITECTURE_X86_32 running on 32 bits.
On this PR I'm adding a condition for the REALM_WATCHOS and architecture REALM_ARCHITECTURE_ARM_64 for watchOS simulators on M1 machines and REALM_WATCHOS combined with REALM_ARCHITECTURE_X86_64 to support Intel simulators running on 64 bits, after some user reported the same error (thread_local not supported).
I tried to find a way to define a variable that let us know if thread_local is available which will make easier the condition here, but didn't find anything. 
This branch is tested by several user having this problem, issues are linked on the description of the pull request.
I'm open to suggestion how we can make this easier.